lib/run/task.ts

```diff
@@ -86,10 +115,37 @@ export function createTask<T>(
             });
           }
         },
-        then: (...args) => getHaltPromise().then(...args),
```

Comment:
Note: there is one very subtle change introduced by this PR: the result of these is no longer a `Promise` and, instead, is now a `PromiseLike`. In practice, this *should* work everywhere (the goal of PromiseLike is that they're meant to be supported by everything magically), but it _is_ possible to write code that only works on Promises and not PromiseLike (ex: any code that uses `instanceof Promise`)